pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -623,61 +623,63 @@ func TestFilterRangefeedInReplicationStream(t *testing.T) {
 	skip.UnderRace(t, "multi cluster/node config exhausts hardware")
 
 	ctx := context.Background()
-
-	filterVal := []bool{}
-	var filterValLock syncutil.Mutex
-
 	clusterArgs := base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
 			DefaultTestTenant: base.TestControlsTenantsExplicitly,
 			Knobs: base.TestingKnobs{
 				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
-				DistSQL: &execinfra.TestingKnobs{
-					StreamingTestingKnobs: &sql.StreamingTestingKnobs{
-						BeforeClientSubscribe: func(_ string, _ string, _ span.Frontier, filterRangefeed bool) {
-							filterValLock.Lock()
-							defer filterValLock.Unlock()
-							filterVal = append(filterVal, filterRangefeed)
-						},
-					},
-				},
 			},
 		},
 	}
 
-	server, s, dbA, dbB := setupLogicalTestServer(t, ctx, clusterArgs, 1)
+	server, s, dbs, _ := setupServerWithNumDBs(t, ctx, clusterArgs, 1, 3)
 	defer server.Stopper().Stop(ctx)
 
+	dbA, dbB, dbC := dbs[0], dbs[1], dbs[2]
+
 	dbAURL, cleanup := s.PGUrl(t, serverutils.DBName("a"))
 	defer cleanup()
 	dbBURL, cleanupB := s.PGUrl(t, serverutils.DBName("b"))
 	defer cleanupB()
 
-	var (
-		jobAID jobspb.JobID
-		jobBID jobspb.JobID
-	)
+	var jobAID, jobBID, jobCID jobspb.JobID
 
-	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", dbBURL.String()).Scan(&jobAID)
-	dbB.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab WITH DISCARD = 'ttl-deletes'", dbAURL.String()).Scan(&jobBID)
+	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE b.public.tab ON $1 INTO TABLE a.tab", dbBURL.String()).Scan(&jobAID)
+	dbB.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE a.tab ON $1 INTO TABLE b.tab WITH DISCARD = 'ttl-deletes'", dbAURL.String()).Scan(&jobBID)
+	dbC.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE a.tab ON $1 INTO TABLE c.tab WITH DISCARD = 'all-deletes'", dbAURL.String()).Scan(&jobCID)
+
+	dbA.Exec(t, "INSERT INTO a.tab VALUES (0, 'a'), (1, 'b'), (2, 'c')")
+	dbA.Exec(t, "UPDATE a.tab SET payload = 'x' WHERE pk = 0")
+
+	dbA.Exec(t, "DELETE FROM a.tab WHERE pk = 2")
+
+	dbC.Exec(t, "SET disable_changefeed_replication = true")
```

Comment:
nit: slightly confusing that you're using your conn to dbC. Could you use dbA instead?

Reply:
also,  why do you need to set `disable_changefeed_replication`?

Reply:
sure, I can do it with A, I was being lazy and didn't want to reset it and I'm done with C at this point but I can just use a -- and reset it.

`SET disable_changefeed_replication` is _how_ TTL deletes ask SQL to ask KV to mark their deletes to be excluded from rangefeeds; all txns in a session with this set get the bit (even non-deletes). Despite the name, the filter has nothing to do with TTLs.